### PR TITLE
Fix control-plane module and add byo certs functionality

### PR DESCRIPTION
**Pull request by git-ival**
~~Note: When the `current_password` was set while using rancher version 2.5.x the the bootstrap resource would fail to authenticate and set the desired rancher password. Due to these bootstrap behavior differences between 2.5.x and 2.6.x I have introduced conditional resource creation for the `rancher2_bootstrap` resource.~~

Setting the `current_password` argument to `null` when the rancher version is >= 2.6.x allows for using a single `rancher2_bootstrap` resource again.

* Added byo certs functionality for the control-plane 
* Added `insecure_flag` var to `clusters/` module to enable using the module with rancher servers using self-signed certs

### CODE REVIEW

File: control-plane/modules/aws-k3s/files/k3s-install.sh [hunk 1]
```diff
@@ -6,6 +6,9 @@ export INSTALL_K3S_VERSION='v${install_k3s_version}'
 export INSTALL_K3S_EXEC='%{ if is_k3s_server }${k3s_tls_san} ${k3s_disable_agent} ${k3s_deploy_traefik} %{~ endif ~}${k3s_exec}'
 export K3S_CLUSTER_SECRET='${k3s_cluster_secret}'
 export K3S_DATASTORE_ENDPOINT='${k3s_datastore_endpoint}'
+export CATTLE_BOOTSTRAP_PASSWORD="admin"
```
Comment (dramich) [inline control-plane/modules/aws-k3s/files/k3s-install.sh:9]: This should be a templated value

Comment (git-ival) [reply control-plane/modules/aws-k3s/files/k3s-install.sh:9]: Actually this isnt meant to be there at all, I forgot to push it up.

Comment (git-ival) [reply control-plane/modules/aws-k3s/files/k3s-install.sh]: Removed

**Comment by git-ival** [comment]
Closing, split into PR #36 and #37

**Pull request closed without merging**